test/common/upstream/upstream_impl_test.cc

```diff
@@ -706,6 +707,34 @@ TEST(PrioritySet, Extend) {
   }
 }
 
+// Resolve zero hosts, while using health checking.
+TEST(ClusterMetadataTest, Metadata) {
+  Stats::IsolatedStoreImpl stats;
+  Ssl::MockContextManager ssl_context_manager;
+  auto dns_resolver = std::make_shared<Network::MockDnsResolver>();
+  NiceMock<Event::MockDispatcher> dispatcher;
+  NiceMock<Runtime::MockLoader> runtime;
+  NiceMock<MockClusterManager> cm;
+  ReadyWatcher initialized;
+
+  const std::string yaml = R"EOF(
+    name: name
+    connect_timeout: 0.25s
+    type: STRICT_DNS
+    lb_policy: ROUND_ROBIN
+    hosts: [{ socket_address: { address: foo.bar.com, port_value: 443 }}]
+  )EOF";
+
+  auto cluster_config = parseClusterFromV2Yaml(yaml);
+  Config::Metadata::mutableMetadataValue(*cluster_config.mutable_metadata(), "com.bar.foo", "baz")
```

Comment:
nit: is it possible to add this to the YAML instead for clarity?

Reply:
Yep, I only did it this way because the yaml for metadata is a little obfuscated/ugly:
```
...
metadata: { filter_metadata: { com.bar.foo: { baz: test_value } } }
...
```
Totally cool doing it that way as long as that seems clearer.

Reply:
It would be clearer to me and I think more instructive for how static config would be written, but I'm fine either way.

Reply:
Gotcha.  That's a good point - I didn't see another example of metadata in yaml or json.